lib/themes/dosomething/paraneue_dosomething/includes/preprocess.inc

```diff
@@ -399,45 +399,72 @@ function paraneue_dosomething_preprocess_reportback_files(&$vars) {
   $initial_count = $reportbacks_gallery['initial_count'];
 
   $reportbacks_gallery['items'] = array();
-   // Set params to retrieve random promoted Reportback Files for this nid.
+
+
+  // Promoted Reportbacks:
+  // ========================================================================
+  // Set params to retrieve random Promoted Reportbacks for this nid.
   $params = array();
   $params['nid'] = $vars['nid'];
   $params['status'] = 'promoted';
   $params['random'] = TRUE;
   $promoted_results = dosomething_reportback_get_reportback_files_query_result($params, $initial_count);
+
   foreach ($promoted_results as $rb_file) {
     $rb_file->image = dosomething_image_get_themed_image_url_by_fid($rb_file->fid, '400x400');
     $reportbacks_gallery['items'][] = paraneue_dosomething_get_gallery_item((array) $rb_file, 'photo');
   }
 
-  // Do we have enough promoted items?
-  $current_count = count($reportbacks_gallery['items']);
-  if ($current_count < $initial_count) {
-    // Find out how many additonal approved Reportback items to grab to fulfill the initial count.
-    $approved_count = $initial_count - $current_count;
-    $reportbacks_gallery['prefetched'] = $approved_count;
-    // Set params to retrieve approved Reportback Files for this nid.
-    $params['status'] = 'approved';
-    unset($params['random']);
-    $approved_results  = dosomething_reportback_get_reportback_files_query_result($params, $approved_count);
-    foreach ($approved_results as $rb_file) {
-      // @todo: DRY. Potentially grab this data from the API endpoint to truly DRY?
-      $rb_file->image = dosomething_image_get_themed_image_url_by_fid($rb_file->fid, '300x300');
-      $reportbacks_gallery['items'][] = paraneue_dosomething_get_gallery_item((array) $rb_file, 'photo');
-    }
-  } else {
-    $reportbacks_gallery['prefetched'] = 0;
-  }
+  // Total number of Promoted Reportbacks obtained.
+  $reportbacks_gallery['total_promoted'] = count($reportbacks_gallery['items']);
+
 
-  // Get the total number of approved reportbacks.
+  // Approved Reportbacks:
+  // ========================================================================
+  // Set params to get the total number of Approved Reportbacks for this nid.
   $params['status'] = 'approved';
   if (isset($params['random'])) {
     unset($params['random']);
   }
+  $reportbacks_gallery['total_approved'] = dosomething_reportback_get_reportback_files_query_count($params);
```

Comment:
This is fine for now, but I've been thinking it may make sense to start storing totals in cache somewhere, so we're not constantly counting totals. I need similar functionality for the RB Reviewer app.  This more of a note than anything to change now though.